File: apps/web/repositories/mock/mock-market-repository.ts
import type { MarketRepository } from "@/repositories/market-repository";
import { MARKET_INDICES } from "@/repositories/mock/mock-source";
import type { MarketIndex, MarketStatus } from "@/types/market";
import { getMockMarketStatus } from "@/utils/market-status";

export class MockMarketRepository implements MarketRepository {
  async getMarketStatus(): Promise<MarketStatus> {
    return getMockMarketStatus();
  }

  async getMarketIndices(): Promise<MarketIndex[]> {
    return MARKET_INDICES;
  }
}

File: apps/api/app/repositories/mock_market_repository.py
from app.repositories.market_repository import MarketRepository
from app.repositories.mock_data import get_market_indices, get_market_status
from app.schemas.market import MarketIndex, MarketStatus


class MockMarketRepository(MarketRepository):
    def get_market_status(self) -> MarketStatus:
        return get_market_status()

    def get_market_indices(self) -> list[MarketIndex]:
        return get_market_indices()

